src/coreclr/jit/importer.cpp

```diff
@@ -8863,7 +8891,7 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
                 {
                     if (opts.IsReadyToRun())
                     {
-                        noway_assert(callInfo->kind == CORINFO_CALL);
+                        // noway_assert(callInfo->kind == CORINFO_CALL);
```

Comment:
I don't think it will build, I found a potential fix I applied https://github.com/dotnet/runtime/pull/52156/commits/65a9154a5d4a2d8518022c58535d8a5a0c169872 but AFAIR it also didn't work as expected for me.

Reply:
It buillt fine without failing the assert (`./build.sh packs` was failing earlier on crossgen2 step), I wanted to see how it plays out in the CI. Looks like `GitHub_36614` regression test failed this assert in checked builds:
https://github.com/dotnet/runtime/blob/ad9979381706d1976e4d26e13cb88d6147264968/src/coreclr/vm/jitinterface.cpp#L10638 so we need to keep this assert.

I wonder if this condition https://github.com/dotnet/runtime/blob/0109769f38670f40856890f69b5652df2e38b438/src/coreclr/jit/importer.cpp#L8890 is `false` on main and for some reason `true` on PR branch? Need to dig deeper. I added `[Intrinsic]` only on GetStubContextAddr and NextCallReturnAddress (the rest were already decorated), so it maybe because of that?

Reply:
Found these diffs with `DOTNET_JitDump=* ./corerun ...`:
```diff
- main
+ pr

# this is ok (old to new style)
-Intrinsic CORINFO_INTRINSIC_Object_GetType Recognized
+Named Intrinsic System.Object.GetType: Recognized

# this is strange..
-Named Intrinsic System.Type.GetTypeFromHandle: Recognized
+Named Intrinsic System.Type.GetTypeFromHandle: Not recognized
-Named Intrinsic System.Type.op_Equality: Recognized
+Named Intrinsic System.Type.op_Equality: Not recognized
```

Reply:
> I applied 65a9154 but AFAIR it also didn't work as expected for me.

Your patch did the trick, thanks! :muscle: (I adapted it a bit as `FEATURE_READYTORUN_COMPILER` has been removed since).

I had a typo in a condition `== 0` was missing from `if (strcmp(className, "RuntimeTypeHandle") == 0)` which was causing unrecognized intrinsic, now all good and green. :christmas_tree: